bindings/c/src/workspace.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw/experimental/workspace.h"
+
+VW_DLL_PUBLIC VWStatus vw_create_workspace(VWOptions* options_handle, VWTraceMessageFunc trace_listener,
+    void* trace_context, VWWorkspace** output_handle, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_create_workspace_with_model(VWOptions* options_handle, void* context, VWReadFunc* read_func,
+    VWTraceMessageFunc* trace_listener, void* trace_context, VWWorkspace** output_handle,
+    VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_create_seeded_workspace(const VWWorkspace* existing_workspace_handle,
+    VWOptions* extra_options_handle, VWTraceMessageFunc* trace_listener, void* trace_context,
+    VWWorkspace** output_handle, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_workspace_get_model_id(
+    const VWWorkspace* workspace_handle, const char** model_id, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_workspace_set_model_id(
+    VWWorkspace* workspace_handle, const char* model_id, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_workspace_get_command_line(
+    const VWWorkspace* workspace_handle, VWString* command_line, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_workspace_finish(VWWorkspace* workspace_handle, VWErrorString* err_str_container) noexcept
+{
+  return VW_not_implemented;
+}
+
+VW_DLL_PUBLIC VWStatus vw_destroy_workspace(VWWorkspace* workspace_handle, VWErrorString* err_str_container) noexcept
```

Comment:
and `vw_workspace_destroy` again for consistency?

Reply:
I think the idea was for create and destroy to be `vw_create_X` and `vw_destroy_X`, but for all of the "member" functions to be `VW_X_VERB`. Do you think we should just opt for the consistency of `VW_ X_VERB` for everything?

Reply:
oh I see it now. I personally like `VW_X_VERB` better but I don't mind `vw_create_X` and `vw_destroy_X`

Reply:
The `create` and `destroy` functions are a bit of a special case in my mind. Where `vw_X_VERB` is performing an operation on `X`, create and destroy are kind of performing operations on the VW environment itself (creating or destroying an object).

I also think its a little more natural to read
`vw_create_example` and `vw_example_create_feature_space`
vs
`vw_example_create` and `vw_example_feature_space_create`